scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,835 @@
+import jdbm.RecordManager;
+import jdbm.RecordManagerFactory;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 26/03/2018
+ */
+@ActiveObject
+public class AOSynchronization implements RunActive, InitActive, EndActive, SynchronizationInternal {
+
+    private static final Logger logger = Logger.getLogger(AOSynchronization.class);
+
+    private static final String RESULT_IS = ", result is: ";
+
+    private static final String QUOTE = "'";
+
+    private static final String PREVIOUS_VALUE_WAS = ", previous value was ";
+
+    private static final String EXCEPTION_WHEN_EVALUATING_CLOSURE = "Exception when evaluating closure : ";
+
+    private static final String ON_KEY = " on key ";
+
+    private static final String WITH_VALUE = ", with value ";
+
+    private File statusFile;
+
+    private File statusFileDirectory;
+
+    private static final String DEFAULT_STORE_SESSION_NAME = "SchedulerStore";
+
+    public static final String STATUS_RECORD_NAME = "STORE";
+
+    private HashMap<String, Channel> inMemoryChannels;
+
+    private PrimaryHashMap<String, Channel> persistedChannels;
+
+    private Queue<Request> waitUntilQueue = new ArrayDeque<>();
+
+    private CompilerConfiguration compilerConfiguration;
+
+    private GroovyShell shell;
+
+    protected RecordManager recMan;
+
+    public AOSynchronization() {
+
+    }
+
+    public AOSynchronization(String statusFileDirectoryPath) {
+        compilerConfiguration = new CompilerConfiguration();
+        compilerConfiguration.setTargetBytecode(CompilerConfiguration.JDK8);
+        shell = new GroovyShell(this.getClass().getClassLoader(), compilerConfiguration);
+        this.statusFileDirectory = new File(statusFileDirectoryPath);
+        if (!statusFileDirectory.exists()) {
+            statusFileDirectory.mkdirs();
+        }
+        statusFile = new File(statusFileDirectory, DEFAULT_STORE_SESSION_NAME);
+        inMemoryChannels = new HashMap<>();
+
+    }
+
+    private boolean init(boolean firstAttempt) {
+        close();
+        try {
+            logger.info("Loading Record Manager from file : " + statusFile);
+            recMan = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
+            persistedChannels = recMan.hashMap(STATUS_RECORD_NAME);
+            logger.info("Content of persisted store : " + persistedChannels);
+            // This empty loop triggers InvalidClassException in case of serial version uid problems
+            for (Map.Entry<String, Channel> tempChannels : persistedChannels.entrySet())
+                ;
+            recMan.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
+            if (firstAttempt) {
+                logger.error("Error occurred when loading channels database " + statusFile.getAbsolutePath() +
+                             ", now cleaning it and retrying.", e);
+                close();
```

Comment:
'Init' method calls 'close' method. This re-initialization looks messy

Reply:
The goal is to clean the database when a serious error occurs when loading it (for example, when serial version uids are different between the current version and the database version). 
We must close the database before cleaning it.

Don't know how to make this much cleaner, but I can extract the part inside the if(firstattempt) inside a dedicated method with a more understandable name.